File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
import java.util.*;

class Solution {
    public int[] solution(int[] sequence, int k) {
        List<int[]> answerList = new ArrayList<>();
        int left = 0;
        int right = 0;
        int sequence_sum = sequence[0];
        
        if (sequence_sum == k) {
            return new int[]{0, 0};
        }
        
        while (true) {
            if (sequence_sum <= k) {
                right += 1;
                if (right == sequence.length) break;
                sequence_sum += sequence[right];
            } else {
                left += 1;
                sequence_sum -= sequence[left - 1];
            }
            
            if (sequence_sum == k) {
                answerList.add(new int[]{left, right});
            }
        }
        
        // for (int[] arr : answerList) {
        //     System.out.println(Arrays.toString(arr));
        // }
        
        int minDiff = Integer.MAX_VALUE;
        int[] result = null;
        for (int[] answer : answerList) {
            int diff = answer[1] - answer[0];
            if (diff < minDiff) {
                minDiff = diff;
                result = answer;
            }
        }
        
        return result;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    left, right = 0, 0
    sequence_sum = sequence[0]
    
    if sequence_sum == k:
        return [0, 0]
    
    while True:
        if sequence_sum <= k:
            right += 1
            if right == len(sequence):
                break
            sequence_sum += sequence[right]
        else:
            left += 1
            sequence_sum -= sequence[left - 1]
        
        if sequence_sum == k:
            answer.append([left, right])
    
    answer.sort(key = lambda x : (x[1] - x[0], x[0]))
    return answer[0]